ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -19,6 +22,22 @@ class Result(NamedTuple):
     file: Scannable  # filename that was scanned
     scan: ScanResult  # Result of content scan
 
+    def enrich_matches(self, lines: List[Line] = []) -> None:
```

Comment:
Using a mutable object as a default value is a bad practice: if the function ever modifies `lines`, the default will be modified for future calls.

It's safer to change lines to be `Optional[List[Line]] = None` and use `if not lines:` (which is more Pythonic than `if len(lines) == 0:` anyway).